storage/localstore/mode_get.go

```diff
@@ -143,6 +145,101 @@ func (db *DB) updateGC(item shed.Item) (err error) {
 	return db.shed.WriteBatch(batch)
 }
 
+func (db *DB) IsHashPresentInRetrievalDataIndex(hash []byte) bool {
+	var item shed.Item
+	item.Address = make([]byte, len(hash))
+	copy(item.Address[:], hash[:])
+	_, err := db.retrievalDataIndex.Get(item)
+	return !(err != nil)
+}
+
+func (db *DB) IsPinnedFileRaw(hash []byte) (bool, error) {
+	var item shed.Item
+	item.Address = make([]byte, len(hash))
+	copy(item.Address[:], hash[:])
+	i, err := db.pinFilesIndex.Get(item)
+	if err != nil {
+		return false, err
+	}
+	raw := false
+	if i.IsRaw > 0 {
+		raw = true
+	}
+	return raw, nil
+}
+
+func (db *DB) IsFilePinned(hash []byte) bool {
+	isFilePinned := false
+	var item shed.Item
+	item.Address = make([]byte, len(hash))
+	copy(item.Address[:], hash[:])
+	i, err := db.pinFilesIndex.Get(item)
+	if err != nil {
+		return false
+	}
+
+	if bytes.Equal(i.Address, hash) {
+		isFilePinned = true
+	}
+	return isFilePinned
+}
+
+func (db *DB) IsChunkPinned(hash []byte) bool {
+	var item shed.Item
+	item.Address = make([]byte, len(hash))
+	copy(item.Address[:], hash[:])
+	_, err := db.pinIndex.Get(item)
+	return !(err != nil)
+}
+
+func (db *DB) GetPinCounterOfChunk(hash []byte) (uint64, error) {
+	var item shed.Item
+	item.Address = make([]byte, len(hash))
+	copy(item.Address[:], hash[:])
+	i, err := db.pinIndex.Get(item)
```

Comment:
The function returns pinCounter.. not Item..
This is where we need generic get, put etc in localstore.
Otherwise everyone will write their own Gets and Puts.